components/sp-provision-service/src/main/java/com/wso2telco/spprovisionapp/api/ApiCallsInAM.java

```diff
@@ -0,0 +1,244 @@
+            String postResponse;
+
+            while ((output = br.readLine()) != null) {
+                totalOutput.append(output);
+            }
+
+            postResponse = "{\"httpCode\": " + response.getStatusLine().getStatusCode() + ",\"output\": "
+                    + totalOutput.toString() + "}";
+            return postResponse;
+        } catch (IOException ex) {
+            logInstance.error("IOException occured in reading responce from AM APIs:" + ex.toString(), ex);
+            String postResponse = "{\"exception\": " + ex.toString() + "\"}";
+            return postResponse;
+        }
+    }
+
+    /*
+     * Create new User in AM
+     */
+    public String createNewUserInAm(String appName, String firstName, String lastName, String devEmail) throws
+            IOException {
+
+        String url = amUserCreationAPI + "?allFieldsValues=" + firstName + "%7C" + lastName + "%7C" + devEmail
+                + "&username=" + appName + "&password=" + password + "&action=addUser";
+        HttpClient httpClient = getNewHttpClient();
+        String response = postToURLjson(url, httpClient);
+        httpClient.getConnectionManager().shutdown();
+        return response;
+    }
+
+    /*
+     * Login the new User to API
+     */
+    public String loginToAm(String appName) throws IOException {
+        cookieStore = null;
+
+
+        String url = amLoginAPI + "?username=" + appName + "&password="
+                + URLEncoder.encode(password, "UTF-8") + "&action=login";
+        DefaultHttpClient httpClient = getNewHttpClient();
+        String response = postToURLjson(url, httpClient);
+        httpClient.getConnectionManager().shutdown();
+        cookieStore = httpClient.getCookieStore();
+        return response;
+    }
+
+    /*
+     * add subscriptions to the newly created app
+     */
+    /*
+     * Service provider get scope
+     */
+
+    /*
+     * Create application in AM
+     */
+    public String createApplication(String appName, String description, String callback, String tier)
+            throws IOException {
+
+        String url = amAppCreationAPI + "?action=addApplication&application=" + appName + "&tier=" + tier
+                + "&description=" + description + "&callbackUrl=" + callback + "";
+        DefaultHttpClient httpClient = getNewHttpClient();
+        httpClient.setCookieStore(cookieStore);
+        String response = postToURLjson(url, httpClient);
+        httpClient.getConnectionManager().shutdown();
+        return response;
+    }
+
+    public String addSubscritions(String userName, String appName, String apiName, String apiVersion,
+                                  String apiProvider, String tier) {
+
+        try {
+            loginToAm(userName);
+        } catch (IOException ex) {
+            logInstance.error("IO Exception occured when trying to login to the user in Add subscription " +
+                    "process" + ex.toString(), ex);
```

Comment:
llogInstance.error("IO Exception occured when trying to login to the user in Add subscription " +
+                    "process" + ex.getMessage(), ex);